flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,256 @@
+        log.debug('URL %s', request_url)
+        json = OmbiBase.get_json(request_url)        
+		
+        entries = []
+        
+        if config.get('type') == 'movies':
+            for request in json:
+                entry = Entry(title=generate_title(request, config.get('type')),
+                              url='http://www.imdb.com/title/' + request.get('imdbId') +'/',
+                              imdb_id=request.get('imdbId'),
+                              tmdb_id=request.get('theMovieDbId'),
+                              movie_name=request.get('title'),
+                              movie_year=int(request.get('releaseDate')[0:4]),
+                              ombi_request_id=request.get('id'),
+                              ombi_released=request.get('released'),
+                              ombi_status=request.get('status'),
+                              ombi_approved=request.get('approved'),
+                              ombi_available=request.get('available'),
+                              ombi_denied=request.get('denied'))
+                if entry.isvalid():
+                    log.debug('Valid entry %s', entry)
+                    if config.get('only_approved') and not entry.get('ombi_approved'):
+                        log.verbose('Request not approved skipping: %s', entry.get('title'))
+                    else:
+                        if not config.get('include_available') and entry.get('ombi_available'):
+                            log.verbose('Request already available skipping: %s', entry.get('title'))
+                        else:
+                            entries.append(entry)
+                else:
+                    log.error('Invalid entry created? %s', entry)
+                    continue
+            return entries
+           
+        if config.get('type') in ['shows', 'seasons', 'episodes']:
+            log.debug('Looking for series')
+            for request in json:
+                log.debug('Found series: %s', request["title"])
+                if config.get('type') == 'shows':
+                    entry = Entry(title=generate_title(request, config.get('strip_year')),
+                                  url='http://www.imdb.com/title/' + request.get('imdbId') +'/',
+                                  series_name=generate_title(request, config.get('strip_year')),
+                                  tvdb_id=request.get('tvDbId'),
+                                  imdb_id=request.get('imdbId'),
+                                  ombi_status=request.get('status'),
+                                  ombi_request_id=request.get('id'))
+                    if entry.isvalid():
+                        log.debug('Valid entry %s', entry)
+                        entries.append(entry)
+                    else:
+                        log.error('Invalid entry created? %s', entry)
+                    continue                
+                else:
```

Comment:
all of the else clauses are way too long and cumbersome. please break them into separate methods, and also try to take the repeated parts to separate method as well